create_graphics.py

```diff
@@ -108,37 +113,29 @@ def create_zip(png_files, zipf):
         # Wait before trying to obtain the lock on the file
         time.sleep(5)
 
-def gather_gribfiles(cla, fhr, gribfiles):
+def gather_gribfiles(cla, fhr, filename, gribfiles):
 
     ''' Returns the appropriate gribfiles object for the type of graphics being
     generated -- whether it's for a single forecast time or all forecast lead
     times. '''
 
-    # We already checked that the current file exists and is old enough, so
-    # assume that the earlier ones are, too.
-
     filenames = {'01fcst': [], 'free_fcst': []}
 
-    fcst_hours = [int(fhr)]
-    if cla.all_leads and gribfiles is None:
-        fcst_hours = list(range(int(fhr) + 1))
+    fcst_hour = int(fhr)
 
-    for fcst_hour in fcst_hours:
-        filename = os.path.join(cla.data_root,
-                                cla.file_tmpl.format(FCST_TIME=fcst_hour))
-        first_fcst = 6 if cla.images[0] == 'global' else 1
-        if fcst_hour <= first_fcst:
-            filenames['01fcst'].append(filename)
-        else:
-            filenames['free_fcst'].append(filename)
+    first_fcst = 6 if cla.images[0] == 'global' else 1
+    if fcst_hour <= first_fcst:
+        filenames['01fcst'].append(filename)
+    else:
+        filenames['free_fcst'].append(filename)
```

Comment:
This code is now only handling one forecast hour at a time. The loop should take place in the main loop for handling missing files appropriately.